manifests/enable.pp

```diff
@@ -0,0 +1,37 @@
+# Manages the enabling FIPS
+#
+# This module will install fips dracut modules and set the fips
+# boot parameter to 1 so the system will boot into fips mode
+#
+class fips::enable (
+  Boolean $aesni   = $::fips::params::aesni
+) {
+
+  kernel_parameter {
+    'fips':
+      value  => '1',
+      notify => Reboot_notify['fips'];
+      # bootmode => 'normal', # This doesn't work due to a bug in the Grub Augeas Provider
+    'boot':
+      value  => "UUID=${::boot_dir_uuid}",
+      notify => Reboot_notify['fips'];
+      # bootmode => 'normal', # This doesn't work due to a bug in the Grub Augeas Provider
+  }
+
+  package {
```

Comment:
I'm 90% sure that, to not have problems, you're going to have to remove this package and then call the `dracut_rebuild` exec when you disable FIPS mode.